src/database/index.tsx

```diff
@@ -174,6 +174,49 @@ export const getBaselineScenarioRef = async () => {
   return results.docs[0].ref;
 };
 
+export const saveScenario = async (scenario: {}): Promise<void> => {
+  try {
+    // We're cheating here because for the launch we know there is only a
+    // baseline scenario. In subsequent launches, we'll need to pass in
+    // the ID of the specific scenario that we want to save. See:
+    // https://github.com/Recidiviz/covid19-dashboard/issues/186
+    const baselineScenarioRef = await getBaselineScenarioRef();
+
+    if (!baselineScenarioRef) return;
+
+    const payload = buildUpdatePayload(scenario);
+
+    return await baselineScenarioRef.update(payload);
+  } catch (error) {
+    console.error("Encountered an error while saving the scenario:");
+    console.error(error);
+  }
+};
+
+export const getBaselineScenario = async (
+  baselineScenarioRef: firebase.firestore.DocumentReference | null,
+) => {
+  if (!baselineScenarioRef) return null;
+
+  const result = await baselineScenarioRef.get();
+  let scenario: Scenario = result.data() as Scenario;
+
+  // NOTE: We should be able to remove this check and save once this issue
+  // is resolved: https://github.com/Recidiviz/covid19-dashboard/issues/186
+  if (scenario && !scenario.hasOwnProperty("promoStatuses")) {
+    scenario = {
+      ...scenario,
```

Comment:
Since this is a "shallow" update (i.e. the `promoStatuses` lives at the "root" level of the Scenario document), you shouldn't need to copy all of the scenario's existing attributes over with the `...scenario`.  I believe the code here can just be:

```javascript
if (scenario && !scenario.hasOwnProperty("promoStatuses")) {
  await saveScenario({
    promoStatuses: {
      dataSharing: true,
      dailyReports: true,
      addFacilities: true,
    }
  });
}
```

Let me know if that doesn't make sense

Reply:
Oh okay -- I was also using that `scenario` variable to return from this function to then set the internal state for the component. Do you recommend still building that object here to return? I had originally tried to return the scenario from `await saveScenario(scenario)`, but ran into some issues with TypeScript since `saveScenario`'s return is a `Promise<void>` and `void` was incompatible with `Scenario` type.

Reply:
Oh, I gotcha.  This additionally merges in the `promoStatus` to the object that gets returned.  Makes sense.  No need to change what you've got.